docs/walletkit/android/chain-abstraction.mdx

```diff
@@ -0,0 +1,233 @@
+              //Chain abstraction is not required, handle transaction as usual
+          }
+        }
+      },
+      onError = { fulfilmentError ->
+            // One of the possible errors: NoRoutesAvailable, InsufficientFunds, InsufficientGasFunds - wallet should send the JsonRpcError to a dapp for given request and display error to the user
+      }
+   )
+  } else {
+    // Handle transaction as usual
+  }
+}
+```
+## Types
+
+Following are the types that are used in the chain abstraction methods.
+
+```kotlin
+data class Transaction(
+  var from: String,
+  var to: String,
+  var value: String,
+  var gas: String,
+  var gasPrice: String,
+  var data: String,
+  var nonce: String,
+  var maxFeePerGas: String,
+  var maxPriorityFeePerGas: String,
+  var chainId: String
+)
+
+sealed class FulfilmentSuccess : Model() {
+  data class Available(
+    val fulfilmentId: String,
+    val checkIn: Long,
+    val transactions: List<Transaction>,
+    val initialTransaction: Transaction,
+    val initialTransactionMetadata: InitialTransactionMetadata,
+    val funding: List<FundingMetadata>
+  ) : FulfilmentSuccess()
+
+  data class NotRequired(val initialTransaction: Transaction) : FulfilmentSuccess()
+}
+
+data class InitialTransactionMetadata(
+  var symbol: String,
+  var amount: String,
+  var decimals: Int,
+  var tokenContract: String,
+  var transferTo: String
+)
+
+data class FundingMetadata(
+  var chainId: String,
+  var tokenContract: String,
+  var symbol: String,
+  var amount: String,
+  var bridgingFee: String,
+  var decimals: Int
+)
+
+sealed class FulfilmentError : Model() {
+  data object NoRoutesAvailable : FulfilmentError()
+  data object InsufficientFunds : FulfilmentError()
+  data object InsufficientGasFunds : FulfilmentError()
+  data class Unknown(val message: String) : FulfilmentError()
+}
+
+sealed class FulfilmentStatus : Model() {
+  data class Completed(val createdAt: Long) : FulfilmentStatus()
+  data class Error(val reason: String) : FulfilmentStatus()
+}
+
+sealed class FulfilmentStatus : Model() {
+  data class Completed(val createdAt: Long) : FulfilmentStatus()
+  data class Error(val reason: String) : FulfilmentStatus()
+}
+
+data class FulfilmentDetails(
+  var fulfilmentDetails: List<TransactionDetails>,
+  var initialDetails: TransactionDetails,
+  var bridgeDetails: List<TransactionFee>,
+  var localTotal: Amount
+)
+
+data class TransactionDetails(
+  var transaction: Transaction,
+  var eip1559: EstimatedFees,
+  var transactionFee: TransactionFee
+)
+
+data class TransactionFee(
+  var fee: Amount,
+  var localFee: Amount
+)
+
+data class Amount(
+  var symbol: String,
+  var amount: String,
+  var unit: String,
+  var formatted: String,
+  var formattedAlt: String
+)
+```
+
+{/* TODO: update sample wallet link */}
+For implementation examples, check our [sample wallet implementation](https://github.com/WalletConnect/WalletConnectKotlinV2/tree/master/sample/wallet) with Kotlin. 
```

Comment:
I'll provide links to classes where the CA is implemented in Sample Wallet 👌

Reply:
@jakubuid can you provide links

Reply:
Here you go:
- https://github.com/reown-com/reown-kotlin/blob/develop/sample/wallet/src/main/kotlin/com/reown/sample/wallet/ui/routes/dialog_routes/session_request/chain_abstraction/ChainAbstractionViewModel.kt#L65
- https://github.com/reown-com/reown-kotlin/blob/develop/sample/wallet/src/main/kotlin/com/reown/sample/wallet/domain/ChainAbstractionUtils.kt